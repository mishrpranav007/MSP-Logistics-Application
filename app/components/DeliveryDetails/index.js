import React from 'react';
import {
  StyleSheet,
  ImageBackground,
  View,
  Text,
  Image,
  TouchableOpacity,
  Platform
} from 'react-native';
import { injectIntl } from 'react-intl';
import deliverydetailsimage from 'app/assets/images/Deliverydetails.png';
import PropTypes from 'prop-types';
import Mask from 'react-native-mask';
import addimage from 'app/assets/images/addimage.png';
import Style from '../../style/index';

const DeliveryDetails = ({ intl, navigation }) => (
  <ImageBackground
    resizeMode="cover"
    source={deliverydetailsimage}
    style={styles.backgroundContainer}
  >
    <View style={styles.textContainer}>
      <Text style={styles.textView}>Jhon Smith</Text>
      <View style={styles.textUnderlineView}></View>
      <Text style={styles.deliveryDetailsText}>Delivery Details</Text>
      <View style={styles.borderLine}></View>
      <View style={styles.RectangleShapeView}>
        <Text style={styles.customerText}>Customer Name</Text>
        <Text style={styles.customernameText}>Sarah Jane</Text>
      </View>

      <View style={styles.shapeSize}>
        <Mask shape="rounded">
          <Image source={addimage} style={styles.addimageview}></Image>
          <Text style={styles.maskedText}>Add Bag photo</Text>
        </Mask>
      </View>

      <View>
        <TouchableOpacity
          style={styles.markdeliveredButton}
          onPress={() => navigation.navigate('DeliveryScreen')}
        >
          <Text style={styles.text}>
            {intl.formatMessage({ id: 'mark_delivered' })}
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  </ImageBackground>
);
const styles = StyleSheet.create({
  backgroundContainer: {
    display: 'flex',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  textContainer: {
    display: 'flex',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
    marginTop: Style.em(3)
  },
  textView: {
    color: Style.WHITE_COLOR,
    height: Style.em(0.813),
    marginRight: Style.em(1)
  },
  textUnderlineView: {
    height: Style.em(0),
    width: Style.em(4.1),
    borderTopColor: Style.WHITE_COLOR,
    borderTopWidth: 1,
    marginTop: Style.em(0.3),
    marginRight: Style.em(1)
  },
  deliveryDetailsText: {
    color: Style.WHITE_COLOR,
    fontSize: Style.em(1.125),
    marginTop: Style.em(1.8),
    marginLeft: Style.em(1),
    marginRight: Style.em(12.5)
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  RectangleShapeView: {
    width: Style.em(19.5),
    height: Style.em(4.5),
    marginTop: Style.em(2),
    marginRight: Style.em(0.9),
    borderRadius: 8,
    backgroundColor: Style.WHITE_COLOR
  },
  customerText: {
    fontSize: Style.em(1),
    marginLeft: Style.em(1),
    marginTop: Style.em(0.2),
    textDecorationColor: Style.GRAY_COLOR
  },
  customernameText: {
    fontSize: Style.em(1.5),
    marginLeft: Style.em(1),
    marginTop: Style.em(0.8)
  },
  borderLine: {
    borderBottomColor: Style.WHITE_COLOR,
    borderBottomWidth: 1
  },
  shapeSize: {
    justifyContent: 'flex-end',
    alignItems: 'center',
    height: Style.em(19.5),
    width: Style.em(19),
    borderRadius: Style.em(0.8),
    marginTop: Style.em(2),
    marginRight: Style.em(1.3),
    backgroundColor: Style.POWDER_COLOR
  },
  addimageview: {
    marginLeft: Style.em(1.2)
  },
  maskedText: {
    marginBottom: Style.em(1.2),
    fontSize: Style.em(1.2),
    textDecorationColor: Style.PRIMARY_COLOR,
    marginTop: Style.em(5.2)
  },
  markdeliveredButton: {
    width: Style.em(20),
    height: Style.em(3.6),
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Style.PRIMARY_COLOR,
    marginTop: Platform.OS === 'ios' ? Style.em(2) : Style.em(2),
    marginRight: Style.em(0.8),
    borderRadius: Style.em(0.69),
    marginBottom: Platform.OS === 'ios' ? Style.em(3.2) : Style.em(5),
    borderTopColor: Style.SECONDARY_COLOR,
    zIndex: 100
  },
  text: {
    color: Style.WHITE_COLOR,
    fontSize: Style.em(1.12),
    textAlign: 'center'
  }
});
DeliveryDetails.propTypes = {
  intl: PropTypes.object,
  navigation: PropTypes.object
};
export default injectIntl(DeliveryDetails);
