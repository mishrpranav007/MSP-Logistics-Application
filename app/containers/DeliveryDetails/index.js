import React, { Component } from 'react';
import {
  StyleSheet,
  ImageBackground,
  View,
  Text,
  Image,
  TouchableOpacity,
  Platform
} from 'react-native';
import { injectIntl } from 'react-intl';
import deliverydetailsimage from 'app/assets/images/Deliverydetails.png';
import PropTypes from 'prop-types';
import Mask from 'react-native-mask';
import addimage from 'app/assets/images/addimage.png';
import ImagePicker from 'react-native-image-picker';
import Style from '../../style/index';

class DeliveryDetails extends Component {
  constructor(props) {
    super(props);
    this.state = {
      photo: null
    };
  }

  handleChoosePhoto = () => {
    const options = {
      noData: true
    };
    ImagePicker.launchImageLibrary(options, response => {
      if (response.uri) {
        this.setState({ photo: response });
      }
    });
  };

  render() {
    const { photo } = this.state;
    return (
      <ImageBackground
        resizeMode="cover"
        source={deliverydetailsimage}
        style={styles.backgroundContainer}
      >
        <View style={styles.textContainer}>
          <Text style={styles.textView}>Jhon Smith</Text>
          <View style={styles.textUnderlineView}></View>
          <Text style={styles.deliveryDetailsText}>Delivery Details</Text>
          <View style={styles.borderLine}></View>
          <View style={styles.RectangleShapeView}>
            <Text style={styles.customerText}>Customer Name</Text>
            <Text style={styles.customernameText}>Sarah Jane</Text>
          </View>

          <View style={styles.shapeSize}>
            <Mask shape="rounded">
              {photo && (
                <Image
                  style={styles.shapeSizeImage}
                  source={{ uri: photo.uri }}
                />
              )}

              {!photo && (
                <View>
                  <TouchableOpacity onPress={this.handleChoosePhoto}>
                    <Image
                      source={addimage}
                      style={styles.addimageview}
                    ></Image>
                  </TouchableOpacity>

                  <Text style={styles.maskedText}>Add Bag photo</Text>
                </View>
              )}
            </Mask>
          </View>

          <View>
            <TouchableOpacity
              style={styles.markdeliveredButton}
              onPress={() => this.props.navigation.navigate('DeliveryDetails')}
            >
              <Text style={styles.text}>
                {this.props.intl.formatMessage({ id: 'mark_delivered' })}
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </ImageBackground>
    );
  }
}
const styles = StyleSheet.create({
  backgroundContainer: {
    display: 'flex',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  textContainer: {
    display: 'flex',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
    marginTop: Style.em(3)
  },
  textView: {
    color: Style.WHITE_COLOR,
    height: Style.em(0.813),
    marginRight: Style.em(1)
  },
  textUnderlineView: {
    height: Style.em(0),
    width: Style.em(4.1),
    borderTopColor: Style.WHITE_COLOR,
    borderTopWidth: 1,
    marginTop: Style.em(0.3),
    marginRight: Style.em(1)
  },
  deliveryDetailsText: {
    color: Style.WHITE_COLOR,
    fontSize: Style.em(1.125),
    marginTop: Style.em(1.8),
    marginLeft: Style.em(1),
    marginRight: Style.em(12.5)
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  RectangleShapeView: {
    width: Style.em(19.5),
    height: Style.em(4.5),
    marginTop: Style.em(2),
    marginRight: Style.em(0.9),
    borderRadius: 8,
    backgroundColor: Style.WHITE_COLOR
  },
  customerText: {
    fontSize: Style.em(1),
    marginLeft: Style.em(1),
    marginTop: Style.em(0.2),
    textDecorationColor: Style.GRAY_COLOR
  },
  customernameText: {
    fontSize: Style.em(1.5),
    marginLeft: Style.em(1),
    marginTop: Style.em(0.8)
  },
  borderLine: {
    borderBottomColor: Style.WHITE_COLOR,
    borderBottomWidth: 1
  },
  shapeSize: {
    justifyContent: 'flex-end',
    alignItems: 'center',
    height: Style.em(19.5),
    width: Style.em(19),
    borderRadius: Style.em(0.8),
    marginTop: Style.em(2),
    marginRight: Style.em(1.3),
    backgroundColor: Style.POWDER_COLOR
  },
  addimageview: {
    marginLeft: Style.em(1.2)
  },
  maskedText: {
    marginBottom: Style.em(1.2),
    fontSize: Style.em(1.2),
    textDecorationColor: Style.PRIMARY_COLOR,
    marginTop: Style.em(5.2)
  },
  markdeliveredButton: {
    width: Style.em(20),
    height: Style.em(3.6),
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Style.PRIMARY_COLOR,
    marginTop: Platform.OS === 'ios' ? Style.em(2) : Style.em(2),
    marginRight: Style.em(0.8),
    borderRadius: Style.em(0.69),
    marginBottom: Platform.OS === 'ios' ? Style.em(3.2) : Style.em(5),
    borderTopColor: Style.SECONDARY_COLOR,
    zIndex: 100
  },
  text: {
    color: Style.WHITE_COLOR,
    fontSize: Style.em(1.12),
    textAlign: 'center'
  },
  shapeSizeImage: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
    width: Style.em(19.5),
    height: Style.em(19)
  }
});
DeliveryDetails.propTypes = {
  intl: PropTypes.object,
  navigation: PropTypes.object
};
export default injectIntl(DeliveryDetails);
