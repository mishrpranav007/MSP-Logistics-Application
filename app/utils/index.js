import { Platform, Dimensions } from 'react-native';

export const isAndroid = () => Platform.OS !== 'ios';
export const getScreenWidth = () => Dimensions.get('window').width;
export const getScreenHeight = () => Dimensions.get('window').height;

export const getPlatformString = () => (isAndroid() ? 'android' : 'ios');
export const getFontFamily = font =>
  isAndroid() ? `Futura-${font}` : `FuturaStd-${font}`;

export const isScreenSmall = () => getScreenHeight() <= 550;
