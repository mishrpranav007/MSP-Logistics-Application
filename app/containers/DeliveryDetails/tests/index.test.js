import React from 'react';
import { renderWithIntl } from 'app/utils/testUtils';
import DeliveryDetails from '../index';

describe('<PickupDetails />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<DeliveryDetails />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 button component inside the Deliverydetailsscreen', () => {
    const { getAllByTestId } = renderWithIntl(<DeliveryDetails />);
    expect(getAllByTestId('delivery-details-button').length).toBe(1);
  });
});
