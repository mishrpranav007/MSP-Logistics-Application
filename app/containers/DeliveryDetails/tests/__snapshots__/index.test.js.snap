// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PickupDetails /> should render and match the snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../app/assets/images/Deliverydetails.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "flex": 1,
            "justifyContent": "center",
            "marginTop": 75.42857142857143,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(255,255,255)",
              "height": 25.142857142857142,
              "marginRight": 37.714285714285715,
              "marginTop": 32.68571428571428,
            }
          }
        >
          Jhon Smith
        </Text>
        <View
          style={
            Object {
              "borderTopColor": "rgb(255,255,255)",
              "borderTopWidth": 1,
              "height": 0,
              "marginRight": 37.714285714285715,
              "marginTop": 10.057142857142857,
              "width": 103.08571428571427,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "rgb(255,255,255)",
              "fontSize": 31.428571428571427,
              "marginLeft": 25.142857142857142,
              "marginRight": 301.7142857142857,
              "marginTop": 20.114285714285714,
            }
          }
        >
          Delivery Details
        </Text>
        <View
          style={
            Object {
              "borderBottomColor": "rgb(255,255,255)",
              "borderBottomWidth": 1,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "rgb(255,255,255)",
              "borderRadius": 8,
              "height": 125.71428571428571,
              "marginRight": 22.628571428571426,
              "marginTop": 30.17142857142857,
              "width": 490.2857142857143,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 25.142857142857142,
                "marginLeft": 25.142857142857142,
                "marginTop": 20.114285714285714,
                "textDecorationColor": "rgb(109,109,109)",
              }
            }
          >
            Customer Name
          </Text>
          <Text
            style={
              Object {
                "fontSize": 45.25714285714285,
                "marginLeft": 25.142857142857142,
                "marginTop": 2.5142857142857142,
                "textDecorationColor": "rgb(109,109,109)",
              }
            }
          >
            Sarah Jane
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(200,247,254)",
              "borderRadius": 20.114285714285714,
              "height": 502.85714285714283,
              "justifyContent": "flex-end",
              "marginRight": 32.68571428571428,
              "marginTop": 40.22857142857143,
              "width": 477.7142857142857,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 0,
                  "overflow": "hidden",
                },
                Object {
                  "height": undefined,
                  "width": undefined,
                },
                Object {
                  "borderRadius": 8,
                },
              ]
            }
          >
            <View>
              <TouchableOpacity
                activeOpacity={0.2}
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../app/assets/images/addimage.png",
                    }
                  }
                  style={
                    Object {
                      "marginLeft": 30.17142857142857,
                    }
                  }
                />
              </TouchableOpacity>
              <Text
                style={
                  Object {
                    "fontSize": 30.17142857142857,
                    "marginBottom": 30.17142857142857,
                    "marginTop": 130.74285714285713,
                    "textDecorationColor": "rgb(36,106,116)",
                  }
                }
              >
                Add Bag photo
              </Text>
            </View>
          </View>
        </View>
        <View>
          <TouchableOpacity
            activeOpacity={0.2}
            disabled={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(36,106,116)",
                "borderRadius": 17.348571428571425,
                "borderTopColor": "rgba(19,43,46,0.83)",
                "elevation": 20,
                "height": 90.5142857142857,
                "justifyContent": "center",
                "marginBottom": 80.45714285714286,
                "marginRight": 20.114285714285714,
                "marginTop": 50.285714285714285,
                "shadowColor": "rgb(255,255,255)",
                "shadowOffset": Object {
                  "height": 10,
                  "width": 0,
                },
                "shadowOpacity": 0.35,
                "shadowRadius": 13.16,
                "width": 502.85714285714283,
                "zIndex": 100,
              }
            }
            testID="delivery-details-button"
          >
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontSize": 28.160000000000004,
                  "textAlign": "center",
                }
              }
            >
              Mark as delivered
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  </View>
</View>
`;
